content/docs/querying/operators.md

```diff
@@ -50,31 +58,101 @@ either `0` (`false`) or `1` (`true`), depending on the comparison result.
 
 `vector1 or vector2` results in a vector that contains all original elements
-(labelsets + values) of `vector1` and additionally all elements of `vector2`,
-which don't have matching labelsets in `vector1`. The metric name is carried
-over from the left-hand-side vector in both cases.
+(label sets + values) of `vector1` and additionally all elements of `vector2`,
+which do not have matching label sets in `vector1`.
+
+## Vector matching
+
+Operations between vectors attempt to find a matching element in the right-hand-side
+vector for each entry in the left-hand-side. There are two basic types of
+matching behaviour:
+
+**One-to-one** finds a unique pair of entries from each side of the operation.
+In the default case, that is an operation following the format `vector1 <operator> vector2`.
+Two entries match if they have the exact same set of labels and corresponding values.
+The `on` keyword allows reducing the set of considered labels to a provided list:
+
+    <vector expr> <bin-op> on(<label list>) <vector expr>
+
+_Modified matching methods can only be used for [comparison/filtering](#comparison-/-filter-binary-operators)
+and [arithmetic](#arithmetic-binary-operators) operations._
+
+Example input:
+
+    method:http_errors:rate5m{method="get", code="500"}  1.64
+    method:http_errors:rate5m{method="get", code="404"}  1.968
+    method:http_errors:rate5m{method="put", code="500"}  0.11
+    method:http_errors:rate5m{method="post", code="500"} 0.31
+    method:http_errors:rate5m{method="post", code="404"} 0.72
+
+    method:http_requests:rate5m{method="get"}  65.6
+    method:http_requests:rate5m{method="del"}  5.6
+    method:http_requests:rate5m{method="post"} 20.0
+
+Example query:
+
+    method:http_errors:rate5m{code="500"} / on(method) method:http_requests:rate5m
+
+This returns a result vector that tells us what fraction of requests had an HTTP
+status code of 500 for each method over the last 5 minutes. Without `on(method)` there
+would have been no match as the metrics do not share the same set of labels.
+The entries with methods `put` and `del` have no match and will not show up in the result:
+
+	{method="get"}  0.025 			//  1.64 / 65.6
+	{method="post"} 0.0155 			//  0.31 / 20.0
+
+**Many-to-one** and **one-to-many** matchings refer to the case that each vector entry on
+the "one"-side can match with numerous elements on the "many"-side. This has to
+be explicitly requested using the `group_left` or `group_right` modifier, where
+left/right determines which vector has the higher cardinality.
+
+    <vector expr> <bin-op> on(<label list>) group_left(<label list>) <vector expr>
+    <vector expr> <bin-op> on(<label list>) group_right(<label list>) <vector expr>
+
+The label list provided with the modifier contains additional labels to be included in
+the result metrics. The intersection of both lists must be empty. Labels appearing on
+either side of the operation may be used but they must not occur on both. Every time
```

Comment:
I found that statement hard to figure out, and it doesn't quite capture the semantics.